src/Common/Core/Impl/OS/Win32EnvironmentBlock.cs

```diff
@@ -16,6 +17,12 @@ public class Win32EnvironmentBlock {
 
         private ConcurrentDictionary<string, string> _environment;
 
+        public IEnumerable<string> EnvironmentVariableNames {
```

Comment:
It seems like it would be better to have the class itself enumerable, and delegate to `_environment.GetEnumerator()`. Then the loop below doesn't have to do lookup for every variable to get the value for its name, it can just iterate over key-value pairs.